vllm/model_executor/layers/quantization/awq_triton.py

```diff
@@ -0,0 +1,272 @@
+                                         shifts[None, None, :]).to(torch.int8)
+
+    iweights = iweights.view(iweights.shape[0], -1)
+
+    # iweights = reverse_awq_order(iweights)
+    # return (iweights & 0xF).to(torch.float16)
+
+    zeros = torch.bitwise_right_shift(qzeros[:, :, None],
+                                      shifts[None, None, :]).to(torch.int8)
+
+    zeros = zeros.view(qzeros.shape[0], -1)
+
+    zeros = reverse_awq_order(zeros)
+    iweights = reverse_awq_order(iweights)
+
+    iweights = torch.bitwise_and(iweights, (2**bits) - 1)
+    zeros = torch.bitwise_and(zeros, (2**bits) - 1)
+
+    scales = scales.repeat_interleave(group_size, dim=0)
+    zeros = zeros.repeat_interleave(group_size, dim=0)
+    print(f"awq_dequantize_torch:iweights.shape = {iweights.shape},"
+          f"zeros.shape={zeros.shape}, "
+          f"scales.shape={scales.shape}")
+
+    return (iweights - zeros) * scales
+
+
+def main():
+    use_triton = True
+    use_torch = True
+
+    qweight_rows = 3584
+    qweight_cols = 576
+    group_size = 128
+    small_test_size = True
+    if small_test_size:
+        qweight_rows = 256
+        qweight_cols = 128
+    print(f"qweight_rows = {qweight_rows}, qweight_cols = {qweight_cols}")
+    qweight_dtype = torch.int32
+    scales_rows = qweight_rows // group_size
+    scales_cols = qweight_cols * 8
+    scales_dtype = torch.float16
+    zeros_rows = scales_rows
+    zeros_cols = qweight_cols
+    zeros_dtype = torch.int32
+    split_k_iters = 0
+    thx = 0
+    thy = 0
+    device = 'cuda'
+    torch.manual_seed(0)
+
+    qweight = torch.randint(0,
+                            10000000, (qweight_rows, qweight_cols),
+                            dtype=qweight_dtype,
+                            device=device)
+    scales = torch.rand(scales_rows,
+                        scales_cols,
+                        dtype=scales_dtype,
+                        device=device)
+    zeros = torch.randint(0,
+                          10000000, (zeros_rows, zeros_cols),
+                          dtype=zeros_dtype,
+                          device=device)
+    print(f"zeros.shape = {zeros.shape}")
+    print(f"qweight = {qweight}")
+    if use_triton:
+        iweights_triton = awq_dequantize_triton(qweight, scales, zeros,
+                                                split_k_iters, thx, thy)
+        print(f"Triton result:iweights_triton = {iweights_triton}")
+        print(f"Any infs in triton result? -->"
+              f"{torch.any(torch.isinf(iweights_triton))}")
+
+    if use_torch:
+        iweights_torch = awq_dequantize_torch(qweight, scales, zeros,
+                                              split_k_iters, thx, thy)
+        print(f"Torch result:iweights_torch = {iweights_torch}")
+
+    if use_torch and use_triton:
+        diff = iweights_torch - iweights_triton
+        error = torch.sum(torch.sqrt(diff * diff))
+        print(f"error = {error}")
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
Could you convert this to a separate test file?

Reply:
Thank you for reviewing, but I've moved it over to this one https://github.com/vllm-project/vllm/pull/7386 , instead of two PRs, I'm going going to have one.